### ISSUE TITLE
error in indexNino34 on LANL IC

### ISSUE THREAD

**Issue by xylar** [MPAS-Dev/MPAS-Analysis#269]
When I run on LANL IC the config file `configs/lanl/config.MatchBoth_orig`, I see an error in `indexNino34` as follows:
```
$ cat indexNino34.log 

Plotting Nino3.4 time series and power spectrum....
  Load SST data...

  Reading files:
    mpaso.hist.am.timeSeriesStatsMonthly.0001-01-01.nc through
    mpaso.hist.am.timeSeriesStatsMonthly.0010-12-01.nc
  Compute NINO3.4 index...
 Computing NINO3.4 power spectra...
 Plot NINO3.4 index and spectra...
analysis task indexNino34 failed during run 
Traceback (most recent call last):
  File "/turquoise/usr/projects/climate/xylar/mustang/mpas_work/analysis/add_mpas_climatology_task/mpas_analysis/shared/analysis_task.py", line 315, in run
    self.run_task()
  File "/turquoise/usr/projects/climate/xylar/mustang/mpas_work/analysis/add_mpas_climatology_task/mpas_analysis/ocean/index_nino34.py", line 195, in run_task
    calendar=calendar)
  File "/turquoise/usr/projects/climate/xylar/mustang/mpas_work/analysis/add_mpas_climatology_task/mpas_analysis/ocean/index_nino34.py", line 648, in _nino34_timeseries_plot
    calendar, axis_font, linewidths, maxXTicks)
  File "/turquoise/usr/projects/climate/xylar/mustang/mpas_work/analysis/add_mpas_climatology_task/mpas_analysis/ocean/index_nino34.py", line 719, in _plot_nino_timeseries
    facecolor='red', interpolate=True, linewidth=0)
  File "/usr/projects/climate/SHARED_CLIMATE/anaconda_envs/default-2.7/lib/python2.7/site-packages/matplotlib/pyplot.py", line 2923, in fill_between
    **kwargs)
  File "/usr/projects/climate/SHARED_CLIMATE/anaconda_envs/default-2.7/lib/python2.7/site-packages/matplotlib/__init__.py", line 1710, in inner
    return func(ax, *args, **kwargs)
  File "/usr/projects/climate/SHARED_CLIMATE/anaconda_envs/default-2.7/lib/python2.7/site-packages/matplotlib/axes/_axes.py", line 4832, in fill_between
    x = ma.masked_invalid(self.convert_xunits(x))
  File "/usr/projects/climate/SHARED_CLIMATE/anaconda_envs/default-2.7/lib/python2.7/site-packages/numpy/ma/core.py", line 2388, in masked_invalid
    condition = ~(np.isfinite(a))
TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

Execution time: 0:01:14.28
```
I ran in batch mode and with the `add_mpas_climatology_tasks` branch but I don't think either of those are relevant to the error.  Instead, I suspect the problem might be with a newer library version on LANL IC than on other machines.  I don't see any obvious cause for the error, as the values being passed in should be floating-point numpy arrays, which should work fin in `isfinite`.

**Comment by xylar** [comment]
@vanroekel, is this something you'd have time to look at at some point?  It's not a huge priority but whatever's causing this problem might creep into analysis on other machines down the road so it would be good to nip it in the bud.

**Comment by xylar** [comment]
This error seems to result from time not getting parsed in the observational data set.  I'm trying to figure out why not and what we can do about it.

**Comment by xylar** [comment]
Scratch that, that doesn't seem to be the problem after all...

**Comment by xylar** [comment]
This appears to be related to this issue:
https://github.com/numpy/numpy/issues/5610

**Comment by xylar** [comment]
So I think the issue is that numpy 1.13 treats datetime64 differently from previous versions.  I'm not entirely sure if the `isfinite` issue is a side effect of this or not.

I think the solution here is to convert from datetime64 to days since 0001-01-01 (same as what we do with MPAS output) and plot the x axis the same as for the MPAS data.  However, I'm having trouble converting from datetime64 to datetime.datetime.  All the help I can find suggests that `.astype(datetime.datetime)` should do the trick but I'm just getting longs instead of datetime.datetime objects.

**Comment by pwolfram** [comment]
Hm, so maybe a work-around outside of numpy may make the most sense.  I'm guessing this is a few lines of code as opposed to a one-liner, e.g., a simple casting function that outputs `datetime64` and then casts it into `datetime.datetime`.  I'm happy to help here if needed.

**Comment by xylar** [comment]
There's not a way to cast datetime64[ns] to datetime.datetime, since the latter doesn't have the temporal resolution.  The simplest fix in the end was to *not* parse the time when the observational data set is read in and convert to days since 0001-01-01 as we do for the model data.  This also lets us plot the x-axis the same way for obs and model data.

I'll submit a PR for this in just a minute.

**Comment by pwolfram** [comment]
Thanks, sounds good @xylar.

**Comment by pwolfram** [comment]
See #270

**Comment by xylar** [comment]
Addressed by #270
